File: notification-adapter-in/src/main/java/com/example/notification/adapter/in/rest/dto/RegisterDeviceTokenRequest.java
package com.example.notification.adapter.in.rest.dto;

import com.example.notification.application.port.in.RegisterDeviceTokenUseCase.RegisterCommand;
import com.example.notification.domain.device.DeviceToken.Platform;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record RegisterDeviceTokenRequest(
        @NotBlank @Size(max = 128) String recipientId,
        @NotNull Platform platform,
        @NotBlank @Size(max = 256) String token) {

    public RegisterCommand toCommand() {
        return new RegisterCommand(recipientId, platform, token);
    }
}

File: notification-adapter-in/src/main/kotlin/com/example/notification/adapter/in/rest/dto/RegisterDeviceTokenRequest.kt
package com.example.notification.adapter.`in`.rest.dto

import com.example.notification.application.port.`in`.RegisterDeviceTokenUseCase
import com.example.notification.domain.device.DeviceToken
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

@JvmRecord
data class RegisterDeviceTokenRequest(
    @field:NotBlank @field:Size(max = 128) val recipientId: String,
    @field:NotNull val platform: DeviceToken.Platform?,
    @field:NotBlank @field:Size(max = 256) val token: String,
) {

    fun toCommand(): RegisterDeviceTokenUseCase.RegisterCommand =
        RegisterDeviceTokenUseCase.RegisterCommand(recipientId, platform!!, token)
}
